drivers/i2c/i2c_ll_stm32_v1.c

```diff
@@ -210,24 +210,32 @@ static inline void handle_addr(struct device *dev)
 		}
 	}
 
-	if (data->current.is_write || data->current.len > 2) {
+	if (data->current.is_write) {
 		LL_I2C_ClearFlag_ADDR(i2c);
 		return;
 	}
-	if (data->current.len == 0U) {
+	/* according to STM32F1 errata we need to handle these corner cases in
+	 * specific way.
+	 * Please ref to STM32F10xxC/D/E I2C peripheral Errata sheet 2.14.1
+	 */
+	if (data->current.len == 0U && IS_ENABLED(CONFIG_SOC_SERIES_STM32F1X)) {
```

Comment:
maybe add parenthesis in the first expression in the if statement

Reply:
@ioannisg Sorry, missed the question. I would say it just a corner case for the F1 platform, because nothing no interrupt follows 0 length transaction